### DISCUSSION TITLE
getting started

### DISCUSSION THREAD

**Discussion by icroseland** [choria-io/general#35] [Q&A]
so, we are in the process of upgrading from a fairly ancient puppet 3.8 setup managing Centos 7 hosts to puppet 7 managing Alma * hosts.

So, far so good.. I am now trying to get choria up and running to replace the mcollective.  I am noticing a small problem though.
/usr/bin/mco -> /opt/puppetlabs/puppet/bin/mco the link is to nothing, and I have no idea where to get a modern enough mco.
none of the puppetlabs packages seem to actually provide /opt/puppetlabs/puppet/bin mco.

choria ping works.

I am pretty sure I am totally missing something important..

**Comment by smortex** [comment]
The [choria-mcorpc-support](https://github.com/choria-io/mcorpc-ruby-support) gem provide compatibility with older MCollective (including the `mco(1)` command).  It is managed by the [choria/mcollective_choria](https://forge.puppet.com/modules/choria/mcollective_choria/) puppet module.

Make sure this is part of your Puppetfile and included in `$mcollective::plugin_classes`:

```puppet
    class { 'choria':
      # ...
    }

    class { 'mcollective':
      plugin_classes => [
        # ...
        'mcollective_choria',
      ],
      # ...
    }
```

**Comment by icroseland** [comment]
good news, got it working.. it was mostly a turning the gems into rpms  and using those problem.
